src/compute-client/src/controller/replicated.rs

```diff
@@ -171,6 +262,20 @@ impl<T> ActiveReplicationState<T>
 where
     T: Timestamp + Lattice,
 {
+    fn add_replica(&mut self, id: ReplicaId) {
+        self.replica_ids.insert(id);
+        for uppers in self.uppers.values_mut() {
+            uppers.add_replica(id);
+        }
+    }
+
+    fn remove_replica(&mut self, id: ReplicaId) {
+        self.replica_ids.remove(&id);
+        for uppers in self.uppers.values_mut() {
+            uppers.remove_replica(id);
+        }
+    }
```

Comment:
This seems like a moment where the lower bound might lurch forward, should we drop a laggy / hung replica, and we won't notice until the next update. Maybe that is fine, but also maybe a thing to call out somewhere along the way (in case it becomes not fine in the future, idk).